### PR TITLE
 start

### PR DESCRIPTION
**Pull request by Kosar0111**

### CODE REVIEW

File: react-redux-app/src/App.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import React, { useState } from 'react';
+import Form from './componets/Form/Form';
+import './app.css';
+import Sort from './componets/Sort/Sort';
+import List from './componets/List/List'
+
+function App() {
```
Comment (danya-prohvat) [inline react-redux-app/src/App.js]: const

Comment (Kosar0111) [reply react-redux-app/src/App.js]: I added new changes

File: react-redux-app/src/store/index.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import { configureStore } from "@reduxjs/toolkit";
```
Comment (danya-prohvat) [inline react-redux-app/src/store/index.js]: why such a strange naming? should be like store.js

File: react-redux-app/src/store/listSlice.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+                discription: action.payload.discription,
+                time: new Date().toString(),
+                edit: false
+            })
+        },
+        deleteTheme(state, action) {
+            state.themes = state.themes.filter(theme => theme.id !== action.payload.id)
+        },
+        updateTheme(state, action) {
+            const { title, discription, id } = action.payload
```
Comment (danya-prohvat) [inline react-redux-app/src/store/listSlice.js]: just updateTheme(state, { payload: {title, discription, id }})

File: react-redux-app/src/store/listSlice.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    name: 'themes',
+    initialState: {
+        themes: []
+    },
+    reducers: {
+        saveTheme(state, action) {
+            state.themes.push({
+                id: uuidv4(),
+                title: action.payload.title,
+                discription: action.payload.discription,
```
Comment (danya-prohvat) [inline react-redux-app/src/store/listSlice.js]: don't make grammatical mistakes *description

File: react-redux-app/src/App.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
+import React, { useState } from 'react';
+import { useSelector } from 'react-redux'
+import Form from './componets/Form/Form';
+import './app.css';
+import Sort from './componets/Sort/Sort';
+import List from './componets/List/List'
+
+function App() {
+  const [start, setStart] = useState(true)
+  const themes = useSelector(state => state.themes.themes);
```
Comment (danya-prohvat) [inline react-redux-app/src/App.js]: themes.themes - unclear, come up with some other name pls

File: react-redux-app/src/componets/Sort/Sort.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+                <div className='sort-title'>
+                    <p>Title</p>
+                    <img className='arrow' src={arrow} alt='' />
+                </div>
+                <div className='sort-date'>
+                    <p onClick={dispatch}>Date </p>
+                    <img className='arrow' src={arrow} alt='' />
+                </div>
+            </div>
+            <input className='sort-searce' type='text' name='title' placeholder='Searce the questions' maxLength='120' /><img className='searce' src={searce} alt='' />
```
Comment (danya-prohvat) [inline react-redux-app/src/componets/Sort/Sort.js]: don't forget alt, it's very important for seo

File: react-redux-app/src/componets/Sort/Sort.js [hunk 7]
```diff
@@ -0,0 +1,27 @@
+import React from 'react'
+import './Sort.css'
+import arrow from '../../img/arrow.svg'
+import searce from '../../img/searce.png'
+import { sortDate } from '../../store/listSlice'
+import { useDispatch } from 'react-redux'
+
+const Sort = () => {
+    const dispatch = useDispatch(sortDate)
```
Comment (danya-prohvat) [inline react-redux-app/src/componets/Sort/Sort.js]: it is not right, you call the hook already with some function, you must pass function into dispatch only when call it
const dispatch = useDispatch()

File: react-redux-app/src/componets/Sort/Sort.js [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+    return (
+        <div className='sort'>
+            <div className='sort-list' >
+                <div className='sort-title'>
+                    <p>Title</p>
+                    <img className='arrow' src={arrow} alt='' />
+                </div>
+                <div className='sort-date'>
+                    <p onClick={dispatch}>Date </p>
```
Comment (danya-prohvat) [inline react-redux-app/src/componets/Sort/Sort.js]: () => dispatch(sortDate)

File: react-redux-app/src/componets/ListItem/ListItem.js [hunk 9]
```diff
@@ -0,0 +1,17 @@
+import './ListItem.css'
+import React, { useState } from 'react'
+import FormEdit from '../FormEdit/FormEdit'
+
+const ListItem = ({ theme }) => {
+    const [edit, setEdit] = useState(false)
```
Comment (danya-prohvat) [inline react-redux-app/src/componets/ListItem/ListItem.js]: better [editMode, setEditMode]

File: react-redux-app/src/componets/ListItem/ListItem.js [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+    const click = (edit) => setEdit(!edit);
+
+    return (
+        <div className='theme' onClick={() => click()}>
```
Comment (danya-prohvat) [inline react-redux-app/src/componets/ListItem/ListItem.js]: just onClick={click}

File: react-redux-app/src/componets/ListItem/ListItem.js [hunk 11]
```diff
@@ -0,0 +1,17 @@
+import './ListItem.css'
+import React, { useState } from 'react'
+import FormEdit from '../FormEdit/FormEdit'
+
+const ListItem = ({ theme }) => {
+    const [edit, setEdit] = useState(false)
+    const click = (edit) => setEdit(!edit);
```
Comment (danya-prohvat) [inline react-redux-app/src/componets/ListItem/ListItem.js]: just const click = () => setEdit(!edit); edit is in the global scope

File: react-redux-app/src/componets/FormEdit/FormEdit.js [hunk 12]
```diff
@@ -0,0 +1,71 @@
+import React, { useState } from 'react'
+import bucet from '../../img/bucet.svg'
+import { updateTheme, deleteTheme } from '../../store/listSlice'
+import './FormEdit.css'
+import { useDispatch, useSelector } from 'react-redux'
+
+const FormEdit = (theme) => {
+    const [title, setTitle] = useState('')
+    const [discription, setDiscription] = useState('')
```
Comment (danya-prohvat) [inline react-redux-app/src/componets/FormEdit/FormEdit.js]: grammar mistake

File: react-redux-app/src/componets/FormEdit/FormEdit.js [hunk 13]
```diff
@@ -0,0 +1,71 @@
...
+    const dispatch = useDispatch()
+
+    console.log(theme);
```
Comment (danya-prohvat) [inline react-redux-app/src/componets/FormEdit/FormEdit.js]: when you push code, remove pls all console logs and delete all warnings

File: react-redux-app/src/componets/FormEdit/FormEdit.js [hunk 14]
```diff
@@ -0,0 +1,71 @@
...
+
+
+    const EditTheme = (e) => {
+        e.preventDefault()
```
Comment (danya-prohvat) [inline react-redux-app/src/componets/FormEdit/FormEdit.js]: why do you need it?

File: react-redux-app/src/componets/FormEdit/FormEdit.js [hunk 15]
```diff
@@ -0,0 +1,71 @@
...
+        dispatch(({ title, discription }))
+        setTitle('')
+        setDiscription('')
+    }
+
+    const removeTheme = (id) => {
```
Comment (danya-prohvat) [inline react-redux-app/src/componets/FormEdit/FormEdit.js]: remove id from parameters, global scope

File: react-redux-app/src/store/store.js [hunk 16]
```diff
@@ -0,0 +1,8 @@
+import { configureStore } from "@reduxjs/toolkit";
+import listSlice from './listSlice'
+
+export default configureStore({
+    reducer: {
+        themes: listSlice,
```
Comment (danya-prohvat) [inline react-redux-app/src/store/store.js:6]: better list:listSlice,

File: react-redux-app/src/componets/Sort/Sort.js [hunk 17]
```diff
@@ -0,0 +1,36 @@
...
+                    <img className='arrow' src={arrow} alt='arrow' />
+                </div>
+            </div>
+            <input
+                className='sort-searce'
+                type='text'
+                name='title'
+                placeholder='Searce the questions' maxLength='120'
+                onChange={e => setSearceTitle(e.target.value)}
+            /><img className='searce' src={searce} alt='searce' />
```
Comment (danya-prohvat) [inline react-redux-app/src/componets/Sort/Sort.js]: grammar

File: react-redux-app/src/store/listSlice.js [hunk 18]
```diff
@@ -0,0 +1,65 @@
+import { createSlice } from "@reduxjs/toolkit";
+import { v4 as uuidv4 } from "uuid";
+
+const listSlice = createSlice({
+  name: "lists",
+
+  initialState: {
+    lists: [],
+    editMode: false,
+    new: false,
```
Comment (danya-prohvat) [inline react-redux-app/src/store/listSlice.js]: newItemMode - better, new - its not clear

File: react-redux-app/src/store/listSlice.js [hunk 19]
```diff
@@ -0,0 +1,65 @@
...
+    },
+    newTheme(state) {
+      state.new = !state.new;
+    },
+    toggleEdit(state) {
+      state.editMode = !state.editMode;
+    },
+    updateTheme(state, action) {
+      const themeEdit = state.lists.find(
+        (list) => list.id === action.payload.id
```
Comment (danya-prohvat) [inline react-redux-app/src/store/listSlice.js]: you may not write '()' if the function has only one argument

File: react-redux-app/src/componets/Form/Form.js [hunk 20]
```diff
@@ -0,0 +1,88 @@
...
+        </div>
+
+        <button type="submit" className={"save"} disabled={!formik.isValid}>
+          Save
+        </button>
+      </form>
+    );
+  } else if (editMode) {
+    return <FormEdit />;
+  } else return null;
```
Comment (danya-prohvat) [inline react-redux-app/src/componets/Form/Form.js]: dont return to jsx null, better empty string - return ''

File: react-redux-app/src/componets/FormEdit/FormEdit.js [hunk 21]
```diff
@@ -0,0 +1,100 @@
...
+const FormEdit = () => {
+  const [modalActive, setModalActive] = useState(false);
+  const allTheme = useSelector((state) => state.themes.lists);
+  const editMode = useSelector((state) => state.themes.editMode);
+  const findId = useSelector((state) => state.themes.editThemeId);
+  const dispatch = useDispatch();
+  const toggleEditeForm = () => dispatch(toggleEdit());
+  const editTheme = allTheme.find((el) => el.id === findId);
+  const onSubmit = (values) => {
+    dispatch(updateTheme({ ...values }));
```
Comment (danya-prohvat) [inline react-redux-app/src/componets/FormEdit/FormEdit.js]: why do you need destructuring ??

File: react-redux-app/src/componets/FormEdit/FormEdit.js [hunk 22]
```diff
@@ -0,0 +1,100 @@
...
+    toggleEditeForm();
+  };
+
+  const validationSchema = yup.object({
```
Comment (danya-prohvat) [inline react-redux-app/src/componets/FormEdit/FormEdit.js]: you created already validationSchema in Form.js. take out it pls in separate folder 'helpers', it must be in src

File: react-redux-app/src/componets/FormEdit/FormEdit.js [hunk 23]
```diff
@@ -0,0 +1,100 @@
...
+
+  return (
+    <div className="edit-all">
+      <form
+        onSubmit={formik.handleSubmit}
+        className={editMode ? "form-edit" : "form-edit-hidden"}
+      >
+        <img
+          className="img-bucet-edit"
+          src={bucet}
```
Comment (danya-prohvat) [inline react-redux-app/src/componets/FormEdit/FormEdit.js]: grammar error, bucket

File: react-redux-app/src/componets/Form/Form.js [hunk 24]
```diff
@@ -0,0 +1,88 @@
...
+          className="input-title"
+          type="text"
+          name="title"
+          placeholder="Title the questions"
+          maxLength="120"
+          onBlur={formik.handleBlur}
+          value={formik.values.title}
+          onChange={formik.handleChange}
+        />
+        {formik.errors.title ? (
```
Comment (danya-prohvat) [inline react-redux-app/src/componets/Form/Form.js]: formik.errors.title && <div className="error">{formik.errors.title}</div>

File: react-redux-app/src/componets/Form/Form.js [hunk 25]
```diff
@@ -0,0 +1,88 @@
...
+        <div className="textarea">
+          <textarea
+            className="form-textarea"
+            name="description"
+            value={formik.values.description}
+            onChange={formik.handleChange}
+            maxLength="500"
+            row="25"
+          ></textarea>
+          {formik.errors.description ? (
```
Comment (danya-prohvat) [inline react-redux-app/src/componets/Form/Form.js]: also &&

File: react-redux-app/src/componets/FormEdit/FormEdit.js [hunk 26]
```diff
@@ -0,0 +1,100 @@
...
+        <input
+          className="input-title-edit"
+          type="text"
+          name="title"
+          maxLength="120"
+          onBlur={formik.handleBlur}
+          value={formik.values.title}
+          onChange={formik.handleChange}
+        />
+        {formik.errors.title ? (
```
Comment (danya-prohvat) [inline react-redux-app/src/componets/FormEdit/FormEdit.js]: &&

File: react-redux-app/src/componets/FormEdit/FormEdit.js [hunk 27]
```diff
@@ -0,0 +1,100 @@
...
+        <div className="textarea-edit">
+          <textarea
+            className="form-textarea-edit"
+            name="description"
+            value={formik.values.description}
+            onChange={formik.handleChange}
+            maxLength="500"
+            row="25"
+          ></textarea>
+          {formik.errors.description ? (
```
Comment (danya-prohvat) [inline react-redux-app/src/componets/FormEdit/FormEdit.js]: &&

File: react-redux-app/src/componets/FormEdit/FormEdit.js [hunk 28]
```diff
@@ -0,0 +1,100 @@
...
+            <div className="error-edit">{formik.errors.description}</div>
+          ) : null}
```
Comment (danya-prohvat) [inline react-redux-app/src/componets/FormEdit/FormEdit.js]: dont return to jsx null

File: react-redux-app/src/componets/List/List.js [hunk 29]
```diff
@@ -0,0 +1,33 @@
...
+  return (
+    <div className="list">
+      <button
+        onClick={hiddenForm}
+        className={!addTheme || editMode ? "new-visible" : "new"}
+      >
+        + New
+      </button>
+      {sortTitle.map((theme) => {
+        return <ListItem key={theme.id} {...theme} />;
```
Comment (danya-prohvat) [inline react-redux-app/src/componets/List/List.js]: you may not return(via return) if you dont use if, for and other js exspression

File: react-redux-app/src/App.js [hunk 30]
```diff
@@ -0,0 +1,39 @@
+import React, { useState } from "react";
```
Comment (danya-prohvat) [inline react-redux-app/src/App.js]: all files that return JSX must have the extension .jsx Change this in every file pls

**Review by danya-prohvat — APPROVED** [review]

File: react-redux-app/src/store/listSlice.js [hunk 31]
```diff
@@ -0,0 +1,166 @@
...
+    updateTheme(state, action) {
+      const themeEdit = state.lists.find(list => list.id === action.payload.id);
+      if (themeEdit) {
+        themeEdit.description = action.payload.description;
+        themeEdit.title = action.payload.title;
+      }
+    },
+    sortDate(state) {
+      state.lists.sort((a, b) => b.time.localeCompare(a.time));
+      console.log(state);
```
Comment (danya-prohvat) [inline react-redux-app/src/store/listSlice.js:93]: dont forget console.log when push

**Review by danya-prohvat — APPROVED** [review]

**Pull request merged**